app/controllers/supervisors/base_controller.rb

```diff
@@ -8,7 +8,7 @@ class BaseController < ApplicationController
     protected
 
     def must_be_supervisor
-      unless signed_in? && current_user.roles.includes?('supervisor')
+      unless signed_in_and_confirmed? && current_user.roles.includes?('supervisor')
```

Comment:
Only vaguely related to the changes here, but thinking about @F3PiX idea on solving this problem, this line actually looks a bit weird to me: why should non-signed-in users get the same redirect as signed in users with the wrong role? I'd even say non-signed-in users shouldn't even know that there are such things as supervisors 😉 

Using the normal `authorize_user!` before action and then stripping this down to only
```ruby
def must_be_supervisor
  unless current_user.roles.includes?('supervisor')
    redirect_to root_path, alert: 'Sorry, the dashboard is for supervisors only'
end
```

Is this what you were referring to @F3PiX? It looks more correct in terms of separating concerns in my opinon.

Reply:
👍 Yes, that is exactly what I mean. 🎯  
`signed_in?` is obsolete because a `current_user` _is_ the signed_in user. 
And I think you mean to add the `before_filter: authenticate_user! `